pkg/connectorbuilder/connectorbuilder.go

```diff
@@ -430,8 +476,31 @@ func NewConnector(ctx context.Context, in interface{}, opts ...Opt) (types.Conne
 			ret.m = metrics.New(metrics.NewNoOpHandler(ctx))
 		}
 
+		if b, ok := c.(EventProviderV2); ok {
+			for _, ef := range b.EventFeeds(ctx) {
+				feedData := ef.EventFeedMetadata(ctx)
+				if feedData == nil {
+					return nil, fmt.Errorf("error: event feed metadata is nil")
+				}
+				if err := feedData.Validate(); err != nil {
+					return nil, fmt.Errorf("error: event feed metadata is invalid: %w", err)
```

Comment:
I just noticed, but it'd be nice to try and log the ID or something that narrows down which event feed is invalid.